### PR TITLE
NFC normalize node list for node shaper by default

### PR DESCRIPTION
**Pull request by zauguin**
Somewhat related to/inspired by https://github.com/latex3/latex2e/issues/552 (and similar issues which keep coming up around Latin Modern's combining characters):

Add a font feature (enabled by default) to NFC normalize the input text before shaping with the default shaper, if the font supports the required characters.

  * Currently it is enabled by default. This changes existing documents, but IMO that's not an issue in this case: Since the code never replaces existing glyphs with non-existing glyphs, no documents should change too much. Also most fonts fall in one of two categories: "Latin Modern" like fonts which are basically broken without NFC input (For these documents might change, but it will probably be an improvement) or modern fonts designed with combining characters in mind where these are rendered identically anyway. Also the effect is limited since most input is in NFC anyway. Also it's much more similar to the HarfBuzz in XeTeX (or `harf` mode) anyway.

    Which leaves the question of performance overhead. Benchmarks of `\hbox{\blindtext[10]}` indicate an overhead of around ~20%~ <10%(in my development version, the released `lua-uni-algos` is slower). I'll try to optimize it a bit more, but getting significant improvements seems hard.

    So basically the question is: Is getting more reliable handling of non NFC glyphs worth 10% overhead by default. I think it is especially since our default font really doesn't like combining characters, but I'm not sure.
  * Currently this introduces a dependency on the `lua-uni-algos` package. We could move the functionality into luaotfload, but I would prefer the opposite approach: luaotfload already duplicates quite some `lua-uni-algos` functionality and I would prefer to replace these with calls to `lua-uni-algos` since then they are maintained in a central place where they can also be used by other people (I would probably move the `lua-uni-algos` repo to latex3 then)

Any opinions?

### CODE REVIEW

**Comment by u-fischer** [comment]
`<10%` sounds not so bad. How can one disable it to test the speed difference? And moving `lua-uni-algos` to latex3 sounds like a good idea, it is better to avoid external dependencies here.

**Comment by zauguin** [comment]
@u-fischer The latest commit uses the `normalize` feature, so use `-normalize` to disable. Also use the latest version from the `lua-uni-algos` repo, otherwise it's much slower.
